sisl/geom/flat.py

```diff
@@ -70,3 +70,124 @@ def graphene(bond=1.42, atoms=None, orthogonal=False):
     if atoms is None:
         return honeycomb(bond, Atom(Z=6, R=bond * 1.01), orthogonal)
     return honeycomb(bond, atoms, orthogonal)
+
+
+def _finish_slab(g, size, vacuum):
+    g = g.repeat(size[1], 1).repeat(size[0], 0)
+    if vacuum is not None:
+        g.cell[2, 2] += vacuum
+        g.set_nsc([3, 3, 1])
+    else:
+        g.set_nsc([3, 3, 3])
+    if np.all(g.maxR(True) > 0.):
+        g.optimize_nsc()
+    return g
+
+
+def fcc100(alat, atoms, size=(1, 1, 2), vacuum=None):
+    """ Face-centered cubic (100) surface
```

Comment:
From this documentation it isn't clear to me along which lattice vector the surface is?

Also, since this is a two-layer structure we are in the situation that one might want to end on an A, or a B?
Instead of using `str` as choice, perhaps integers would be ok? so `end_layer=0|1` or something, the same applies to the other surfaces.

Reply:
We could introduce another optional parameter `axis=2` to select along which lattice vector the layers are grown?

Regarding the layer growth one could also have `layer_offset=0` which simply circles through the possibilities for the first growth layer (A/B for 100 and 110 and A/B/C for 111)?

Reply:
Agreed!  

The choice of what to signify as the *important* layer is also not clear. For instance for TS I would mostly like to control the end-surface layer, I don't care too much about the start-surface layer. And when testing different number of layers it seems rather annoying to have to calculate `size` depending on what your initial layer and the number of layers you want. In other cases the initial layer would be important (or just the other electrode).

So perhaps it would be best if one could do something like:
```python
def fcc_slab(alat, atoms, start=None, end=None, ...):
    # possibly layer_start/layer_end
    if start is not None and end is not None:
         raise ValueError("Only one of start or end may be supplied")
```
Then one could easily figure out how to define the full slab?

Reply:
What do you think about `fcc_slab(alat, atoms, miller, start='A', end=None, ...)` as the default? With the idea that either `start` or `end` can be a string `A`, `B`, or `C`?

Reply:
I think allowing `str` and `int` are fine. I probably prefer `int` since it can be negative as well, indicating the *final* of the sequence.
So
```python
# ABC stacking
A == 0 == -3 == 3
B == 1 == -2 == 4
C == 2 == -1 == 5
```
Internal conversion is easy:
```python
if isinstance(start, str):
    start = "ABC".index(start)
if isinstance(start, int):
    start = start % 3
```
and it will always be `0, 1, 2`. I am thinking about programatically creating variations of slabs, and here integers are easier to manage?

Reply:
Done!